### PR TITLE
Feature Request: Added Mish activation function

### PR DESCRIPTION
**Pull request by digantamisra98**
Mish is a new novel activation function proposed in this [paper](https://arxiv.org/abs/1908.08681).
It has shown promising results so far and has been adopted in several packages including:

- [TensorFlow-Addons](https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations)
- [SpaCy (Tok2Vec Layer)](https://github.com/explosion/spaCy)
- [Thinc - SpaCy's official NLP based ML library](https://github.com/explosion/thinc/releases/tag/v7.3.0)
- [Echo AI](https://github.com/digantamisra98/Echo)
- [Eclipse's deeplearning4j](https://github.com/eclipse/deeplearning4j/issues/8417)
- [CNTKX - Extension of Microsoft's CNTK](https://github.com/delzac/cntkx)
- [FastAI-Dev](https://github.com/fastai/fastai_dev/blob/0f613ba3205990c83de9dba0c8798a9eec5452ce/dev/local/layers.py#L441)
- [Darknet](https://github.com/AlexeyAB/darknet/commit/bf8ea4183dc265ac17f7c9d939dc815269f0a213)
- [Yolov3](https://github.com/ultralytics/yolov3/commit/444a9f7099d4ff1aef12783704e3df9a8c3aa4b3)
- [BeeDNN - Library in C++](https://github.com/edeforas/BeeDNN)
- [Gen-EfficientNet-PyTorch](https://github.com/rwightman/gen-efficientnet-pytorch)
- [dnet](https://github.com/umangjpatel/dnet/blob/master/nn/activations.py)

All benchmarks, analysis and links to official package implementations can be found in this [repository](https://github.com/digantamisra98/Mish) 

It would be nice to have Mish as an option within the activation function group.

This is the comparison of Mish with other conventional activation functions in a SEResNet-50 for CIFAR-10:  (Better accuracy and faster than GELU)
se50_1

### CODE REVIEW

**Comment by f0k** [comment]
Hey Diganta! Thank you for your PR! In the Travis log, I noticed that py.test does not run through any more... will look into this. Regarding Mish, PyTorch specifically decided against including it (for now) until there's clear evidence that it is useful. In my own experiments, it worked better than ReLU, but worse than LReLU. Nevertheless, I'm not opposed to including Mish (and Swish) to make it easier to try out. We'd need a test to keep 100% test coverage, though.

**Comment by digantamisra98** [comment]
@f0k Thanks for the response. appreciate your help. Regarding PyTorch denying to include both Mish and Swish is because they haven't been accepted as community standard so getting added to Lasagne is a step closer to allowing more people in the community to give it a try.  Additionally if you can post the logs for your experiment, that would be very helpful.

**Comment by f0k** [comment]
The Travis problem is dealt with in #921.

> getting added to Lasagne is a step closer to allowing more people in the community to give it a try

A rather small step, though -- the user base of Lasagne is considerably smaller now that there are so many deep learning frameworks, and trying mish and swish is easy enough without an implementation provided by a framework.

> Additionally if you can post the logs for your experiment, that would be very helpful.

I think that'll be of limited use. Email me if you'd like a copy of the learning curves, but this was not on a public dataset.

**Comment by digantamisra98** [comment]
> A rather small step, though -- the user base of Lasagne is considerably smaller now that there are so many deep learning frameworks, and trying mish and swish is easy enough without an implementation provided by a framework.

Agreed, however, most of the people getting into this domain have been reluctant to try something which doesn't come out of the box (Personal observation), Also, a small step is a step nonetheless. 

> I think that'll be of limited use. Email me if you'd like a copy of the learning curves, but this was not on a public dataset.

Sure, will drop an email. Thanks!
